plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryPageSink.java

```diff
@@ -64,28 +73,46 @@ public BigQueryPageSink(
         this.pageSinkIdColumnName = requireNonNull(pageSinkIdColumnName, "pageSinkIdColumnName is null");
         checkArgument(temporaryTableName.isPresent() == pageSinkIdColumnName.isPresent(),
                 "temporaryTableName.isPresent is not equal to pageSinkIdColumn.isPresent");
-        this.tableId = temporaryTableName
-                .map(tableName -> TableId.of(remoteTableName.getProjectId(), remoteTableName.getDatasetName(), tableName))
-                .orElseGet(remoteTableName::toTableId);
+        this.tableName = temporaryTableName
+                .map(tableName -> TableName.of(remoteTableName.getProjectId(), remoteTableName.getDatasetName(), tableName))
+                .orElseGet(remoteTableName::toTableName);
     }
 
     @Override
     public CompletableFuture<?> appendPage(Page page)
     {
-        InsertAllRequest.Builder batch = InsertAllRequest.newBuilder(tableId);
+        JSONArray batch = new JSONArray();
         for (int position = 0; position < page.getPositionCount(); position++) {
```

Comment:
should the data be batched based on a request size limit (config)? https://cloud.google.com/bigquery/quotas#write-api-limits

Reply:
good to do but pre-existing issue.